docs/spec_draft.md

```diff
@@ -127,3 +127,183 @@ specification.
 %z = stablehlo.add %x, %y : tensor<2x2xf32>
 // %z: [[6, 8], [10, 12]]
 ```
+
+## stablehlo.and
+
+`stablehlo.and(lhs, rhs) -> result`
+
+### Semantics
+
+An element-wise operation performing bitwise AND of two tensors `lhs` and `rhs`
+of integer types and produces a `result` tensor. For Boolean tensors, it
+computes the logical operation.
```

Comment:
This raises the question of when bitwise AND would be different from logical AND, which raises the question of what's the representation of `pred`. We're talking about representations for integer and floating types, but not booleans.

Reply:
>> what's the representation of `pred`
In the specification, we have representations for ints and floats which just happened to match the actual llvm types we use to implement them. (Side note: The intention is not to give away that implementation info in the spec.) 
Similarly,  we have the Boolean types represented as `pred`. 

Let me know if I am missing anything.

Reply:
"The intention is not to give away that implementation info in the spec". I mean, the spec is fairly precise for the representation of integer and floating-point types, referring to "signed integer with two’s complement representation", "IEEE 754 format", "Brain Floating-Point Format".

Perhaps we should provide more details about booleans as well. E.g. "Boolean types referred to in the document as `pred`. Exact representation of boolean types (e.g. 1 byte per boolean vs 1 bit per boolean) is implementation-define"?

Reply:
I got your point. Added the detail.